crates/sui-indexer-alt-framework/src/pipeline/concurrent/mod.rs

```diff
@@ -75,16 +76,18 @@ pub trait Handler: Processor<Value: FieldCount> {
 
     /// Take a chunk of values and commit them to the database, returning the number of rows
     /// affected.
-    async fn commit(values: &[Self::Value], conn: &mut db::Connection<'_>)
-        -> anyhow::Result<usize>;
+    async fn commit<'a>(
+        values: &[Self::Value],
+        conn: &mut <Self::Store as Store>::Connection<'a>,
```

Comment:
(optional): I did some digging into why this explicit trait constraining is required, it's just a limitation of the type system: https://github.com/rust-lang/rust/issues/38078.

And it's not a big deal in this case (it's not too difficult to understand what's going on), but there was a nice workaround suggested in the issue, which is to introduce a type alias:

```rust
type StoreConnection<'a, S> = <S as Store>::Connection<'a>;
```

Which can be used here (and elsewhere) to refer to the nested associated connection type for a store associated type.